File: TempoLux/backend/src/main/java/com/tempolux/catalog/Watch.java
package com.tempolux.catalog;

import java.math.BigDecimal;

/** Modelo inmutable de un reloj del catálogo. Incluye nombre, descripción y materiales en ES/EN. */
public record Watch(
    int id,
    String slug,
    String imageUrl,
    BigDecimal price,
    String currency,
    LocalizedText name,
    LocalizedText description,
    LocalizedMaterials materials) {
}

File: TempoLux/src/app/watch.ts
/** Idiomas soportados por la aplicación */
export type Language = 'es' | 'en';

/** Par de texto bilingüe para nombres y descripciones */
export interface LocalizedText {
  es: string;
  en: string;
}

/** Lista de materiales bilingüe para mostrar en la tarjeta del catálogo */
export interface LocalizedMaterials {
  es: string[];
  en: string[];
}

/** Modelo de reloj tal como lo devuelve el backend */
export interface Watch {
  id: number;
  slug: string;
  imageUrl: string;
  price: number;
  currency: 'EUR'; // divisa única en esta versión
  name: LocalizedText;
  description: LocalizedText;
  materials: LocalizedMaterials;
}
